### ISSUE TITLE
[Bug]: `isTemplateDeclaration` and `isTemplateDeclarationOrInstance` take `TemplatedType` instead of just `Type`

### ISSUE THREAD

**Issue by TomerAberbach** [microsoft/typespec#5447]
### Describe the bug

`isTemplateDeclaration` and `isTemplateDeclarationOrInstance` take `TemplatedType` instead of just `Type`. This makes them a lot less useful and inconsistent with `isTemplateInstance`, which takes just a `Type`.

Presumably they were meant to take `Type` because they each have a type assertion of `is TemplatedType`, which is redundant with the parameter already being `TemplatedType`.

I ended up having to write this in my code (and checked that first) so that I could pass a type to `isTemplateDeclaration`:
```ts
const isTemplatedType = (type: Type): type is TemplatedType => {
  switch (type.kind) {
    case `Scalar`:
    case `Union`:
    case `Model`:
    case `Interface`:
    case `Operation`:
      return true
    default:
      return false
  }
}
```

### Reproduction

See above

### Checklist

- [x] Follow our [Code of Conduct](https://github.com/microsoft/typespec/blob/main/CODE_OF_CONDUCT.md)
- [x] Check that there isn't already an issue that request the same bug to avoid creating a duplicate.
- [x] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/Microsoft/typespec/discussions).
- [x] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by witemple-msft** [comment]
Good idea. We can just accept `Type` and refine the type internally. I think `isTemplateInstance` was probably added separately from the other two, but there's no reason we can't accept any `Type` as inputs to those instead of requiring you to know that the type _could_ be a template.

Linked PR will resolve this issue.

**Comment by witemple-msft** [comment]
Reopening on reversion of #5527
